### PR TITLE
Updates Prefect 1 Migration Guide

### PR DESCRIPTION
**Pull request by EmilRex**
Updates migration guide from v2 to v3, mainly for accuracy. The initial change does not significantly enhance or alter the material. Links have been updated to point to v3 docs in most cases.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: docs/guides/migration-guide.md [hunk 1]
```diff
@@ -8,128 +8,112 @@ search:
...
 ---
 
-# Migrating from Prefect 1 to Prefect 2
+# Migrating from Prefect 1 to Prefect 3
 
-This guide is designed to help you migrate your workflows from Prefect 1 to Prefect 2. 
+Learn how to migrate your workflows from Prefect 1 to Prefect 3. Check out the [quickstart](https://docs.prefect.io/3.0/get-started/quickstart) for a high level overview.
+
+!!! Note "What about Prefect 2?"
+    Prefect 2 refers to the 2.x lineage of the open source prefect package, and Prefect 3 refers exclusively to the 3.x lineage of the prefect package. Neither version is strictly tied to any aspect of Prefect’s commercial product, [Prefect Cloud](https://docs.prefect.io/3.0/manage/cloud/index).
```
Comment (zzstoatzz) [inline docs/guides/migration-guide.md:16]: perhaps we could add some note about how the jump from 2 to 3 is not nearly as big as 1 to 2

relatedly, I like what (I think) this is going for, which is "cloud works for both 2.x and 3.x, dont worry" but the way this sentence is worded is sort of confusing to me

**Review by zzstoatzz — COMMENTED** [review]
love to see this!

File: docs/guides/migration-guide.md (hunk 1 above)
Comment (kevingrismore) [reply docs/guides/migration-guide.md:16]: How about just

> Prefect Cloud is compatible with all 2.x and 3.x package versions.

File: docs/guides/migration-guide.md [hunk 2]
```diff
@@ -8,128 +8,112 @@ search:
...
-- A first-class `subflows` concept: Prefect 1 only allowed the [flow-of-flows orchestrator pattern](https://discourse.prefect.io/tag/orchestrator-pattern). With Prefect 2 subflows, you gain a natural and intuitive way of organizing your flows into modular sub-components. For more details, see [the following list of resources about subflows](https://discourse.prefect.io/tag/subflows).
-
+- [Blocks](https://docs.prefect.io/3.0/develop/blocks) allow you to securely store UI-editable, type-checked configuration to external systems. All those components are configurable in one place and provided as part of the open-source Prefect 3 product. In contrast, the concept of [Secrets](https://docs-v1.prefect.io/orchestration/concepts/secrets.html) in Prefect 1 are more narrowly scoped and only available in Prefect Cloud.
+- [Automations](https://docs.prefect.io/3.0/automate/events/automations-triggers) provide an expressive API to define triggers (based on events) and actions. Automations power notifications and much more.
+- Every flow state change, task state change, and more produces a corresponding event which can be the basis of an automation trigger.
```
Comment (discdiver) [inline docs/guides/migration-guide.md:94]:
```suggestion
- Every action within Prefect, including flow state changes and task state changes, produces a corresponding event that can trigger an automatic action. Potential actions include running flows, sending notifications, and pausing or restarting schedules.
```

File: docs/guides/migration-guide.md [hunk 3]
```diff
@@ -8,128 +8,112 @@ search:
...
+- First-class support for `subflows` which enables a natural and intuitive way of organizing your flows into modular sub-components. In contrast, Prefect 1 only allowed the [flow-of-flows orchestrator pattern](https://docs-v1.prefect.io/core/idioms/flow-to-flow.html).
```
Comment (discdiver) [inline docs/guides/migration-guide.md:95]:
```suggestion
- First-class support for nested flows enables a natural and intuitive way of organizing your flows into modular sub-components. In contrast, Prefect 1 only allowed the [flow-of-flows orchestrator pattern](https://docs-v1.prefect.io/core/idioms/flow-to-flow.html).
```

File: docs/guides/migration-guide.md [hunk 4]
```diff
@@ -8,128 +8,112 @@ search:
...
-
-### Incremental adoption
-The built-in SQLite database automatically tracks all your locally executed flow runs. As soon as you start a Prefect server and open the Prefect UI in your browser (or [authenticate your CLI with your Prefect Cloud workspace](/ui/cloud/)), you can see all your locally executed flow runs in the UI. You don't even need to start an agent.
-
-Then, when you want to move toward scheduled, repeatable workflows, you can build a deployment and send it to the server by running a CLI command or a Python script. 
-
-- You can create a deployment to on remote infrastructure, where the run environment is defined by a reusable infrastructure block.
+With Prefect 3, your functions *are* your flows and tasks. Prefect 3 automatically detects your flows and tasks without the need to define a rigid DAG structure. While use of tasks is encouraged to provide you the maximum visibility into your workflows, they are no longer required. You can add a single `@flow` decorator to your main function to transform any Python script into a Prefect workflow.
```
Comment (discdiver) [inline docs/guides/migration-guide.md:105]:
```suggestion
With Prefect 3, your decorated functions *are* your flows and tasks. Prefect 3 automatically detects your flows and tasks without the need to define a rigid DAG structure. While use of tasks is encouraged to provide you the maximum visibility into your workflows, they are no longer required. You can add a single `@flow` decorator to your main function to transform any Python script into a Prefect workflow.
```

File: docs/guides/migration-guide.md [hunk 5]
```diff
@@ -8,128 +8,112 @@ search:
   boost: 2
 ---
 
-# Migrating from Prefect 1 to Prefect 2
+# Migrating from Prefect 1 to Prefect 3
```
[the remark this answers is not in the document]
Comment (EmilRex) [reply docs/guides/migration-guide.md:11]: @discdiver just in the title or throughout?

[the remark this answers is not in the document]
Comment (discdiver) [reply docs/guides/migration-guide.md:11]: Throughout would match the v3 guide.

File: docs/guides/migration-guide.md (hunk 1 above)
Comment (EmilRex) [reply docs/guides/migration-guide.md:16]: How about:
> Prefect Cloud is compatible with all 2.x and 3.x package versions. We recommend adopting Prefect 3.0 as it is significantly more performant than Prefect 2.x and includes new features.

By the way, I ripped the original wording directly from: https://docs.prefect.io/3.0/resources/whats-new-prefect-3

Comment (discdiver) [reply docs/guides/migration-guide.md:16]: Sounds good to me!

**Review by discdiver — APPROVED** [review]

**Pull request merged**